lightly/api/api_workflow_client.py

```diff
@@ -220,3 +220,49 @@ def upload_file_with_signed_url(
             response = sess.put(signed_write_url, data=file)
         response.raise_for_status()
         return response
+
+    def set_request_timeout(self, timeout: Union[int, Tuple[int, int]]):
+        """Sets a default timeout for all api requests.
+
+        Args:
+            timeout:
+                Timeout in seconds. Is either a single total_timeout value or a
+                (connect_timeout, read_timeout) tuple. 
+                See https://urllib3.readthedocs.io/en/stable/reference/urllib3.util.html?highlight=timeout#urllib3.util.Timeout
+                for details on the different values.
+        """
+        set_api_client_request_timeout(client=self.api_client, timeout=timeout)
```

Comment:
maybe add a sanity check if `self.api_client` is actually available? (otherwise its strongly dependent on the order of calling these fn)

Reply:
The `self.api_client` attribute should always exist as it is created in the `__init__` method. The only error that can happen is if someone changes the order of the function calls in the init itself but this is covered by unit tests.